File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/exceptions/WalletOpenException.java
package com.evernym.verity.sdk.exceptions;

/**
 * Exception class that is thrown when the wallet fails to open
 */
public class WalletOpenException extends WalletException {
    /**
     * Constructs a new InvalidMessageTypeException with the given message.
     *
     * @param message the given message for the cause of the exception.
     *
     * @see VerityException#VerityException(String)
     */
    public WalletOpenException(String message) {super(message);}

    /**
     * Constructs a new InvalidMessageTypeException with the given message and cause.
     *
     * @param message the given message for the cause of the exception.
     * @param cause the given cause (Throwable) for the exception.
     *
     * @see VerityException#VerityException(String, Throwable)
     */
    public WalletOpenException(String message, Throwable cause) {super(message, cause);}
}

File: sdk/dotnet-sdk/src/VeritySDK/Exception/WalletOpenException.cs
using System;

namespace VeritySDK
{
    ///// <summary>
    ///// Exception class that is thrown when the wallet fails to open
    ///// </summary>
    public class WalletOpenException : WalletException
    {
        /// <summary>
        /// Constructs a new exception with the given message.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        public WalletOpenException(string message) : base(message) { }

        /// <summary>
        /// Constructs a new exception with the given message and cause.
        /// </summary>
        /// <param name="message">the given message for the cause of the exception.</param>
        /// <param name="cause">the given cause for the exception.</param>
        public WalletOpenException(String message, System.Exception cause) : base(message, cause) { }
    }

}
